fs/march/march.go

```diff
@@ -122,6 +122,7 @@ func (m *March) Run() {
 		dstDepth = fs.MaxLevel
 	}
 
+	merr := &multierror{}
```

Comment:
If you used `github.com/uber-go/multierr` you could avoid allocation here.

Reply:
What about concurrent usage needed below?